### PR TITLE
Git Workflow: Add action-tmate to mac git workflow

### PR DESCRIPTION
**Pull request by praveenkumar**
This GitHub Action offers us a direct way to interact with the
host system on which the actual scripts (Actions) will run.

- https://github.com/mxschmitt/action-tmate

### CODE REVIEW

File: .github/workflows/macos-installer.yml [hunk 1]
```diff
@@ -38,6 +38,14 @@ jobs:
...
       - name: Build macOS installer
         run: make NO_CODESIGN=1 out/macos-amd64/crc-macos-amd64.pkg
+      - name: Install crc pkg
+        run: sudo installer -pkg out/macos-amd64/crc-macos-amd64.pkg -target /
+      - name: Set podman preset as part of config
+        run: crc config set preset podman
+      - name: Run crc setup command
+        run: crc setup
+      - name: Run crc with podman preset
+        run: crc start
```
[the remark this answers is not in the document]
Comment (praveenkumar) [reply .github/workflows/macos-installer.yml:48]: @cfergeau I am still in process, I will put hold.

**Comment by cfergeau** [comment]
You can use this kind of changes:
```
diff --git a/.github/workflows/macos-installer.yml b/.github/workflows/macos-installer.yml
index 159d8ff2..66f4f940 100644
--- a/.github/workflows/macos-installer.yml
+++ b/.github/workflows/macos-installer.yml
@@ -4,6 +4,7 @@ on:
     branches:
       - "main"
       - "old-main"
+      - "mac_git_workflow"
   pull_request: {}
 jobs:
   build:
```
in order to get github-actions to get triggered when you push to your own repo/branch without needing a PR

File: .github/workflows/macos-installer.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (praveenkumar) [reply .github/workflows/macos-installer.yml:50]: @cfergeau Github runner uses https://docs.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series for windows/linux and doesn't support the nested virtualisation , it only supported by https://docs.microsoft.com/en-us/azure/virtual-machines/dv3-dsv3-series one so with glthub runner it is not possible to support same.

**Pull request merged**